R/read_redcap.R

```diff
@@ -463,12 +462,21 @@ add_event_mapping <- function(supertbl, linked_arms, repeat_event_types) {
   event_info <- linked_arms
 
   if (!is.null(repeat_event_types)) {
+    # Preserve factor levels post-join by referencing level order from linked_arms
+    repeat_event_types$redcap_event_name <- factor(repeat_event_types$redcap_event_name,
+      levels = levels(event_info$unique_event_name),
+      ordered = TRUE
```

Comment:
I would actually not make this ordered. We don't really know if the events in an arbitrary REDCap have meaningful order and using a plain `factor` will still preserve the order if they are meaningful.

Reply:
Well, for my purposes downstream in `record_status_dashboard()` creation I wanted to capture info on the order that events appear in the display. There's no real assumptions (to my knowledge) I can make otherwise when just accepting the supertibble. I believe the order that events come out resembles how they're ordered in the UI.

Reply:
Confirmed this will work using Prodigy as an example.

```r
> db$redcap_events[[1]]
# A tibble: 1 × 5
  redcap_event      event_name        redcap_arm arm_name repeat_type 
  <ord>             <fct>             <chr>      <chr>    <chr>       
1 infusion_sequence Infusion Sequence 1          Arm 1    nonrepeating
> db$redcap_events[[1]]$event_name
[1] Infusion Sequence
23 Levels: Infusion Sequence Screening & Enrollment Manufacturing LD Chemo Pre-Infusion Infusion 1 Day Follow-Up ... Safety/Meds
> db$redcap_events[[1]]$event_name %>% levels()
 [1] "Infusion Sequence"      "Screening & Enrollment" "Manufacturing"          "LD Chemo"               "Pre-Infusion"          
 [6] "Infusion"               "1 Day Follow-Up"        "3 Day Follow-Up"        "7 Day Follow-Up"        "10 Day Follow-Up"      
[11] "14 Day Follow-Up"       "21 Day Follow-Up"       "28 Day Follow-Up"       "2 Month Follow-Up"      "3 Month Follow-Up"     
[16] "4 Month Follow-Up"      "5 Month Follow-Up"      "6 Month Follow-Up"      "9 Month Follow-Up"      "12 Month Follow-Up"    
[21] "Unscheduled"            "End of Study"           "Safety/Meds"    
```